### PR TITLE
https://wso2.org/jira/browse/BPS-973

### PR DESCRIPTION
**Pull request by NatashaWso2**

### CODE REVIEW

File: components/bpmn/bpmn-rest/src/main/java/org/wso2/carbon/bpmn/rest/service/stats/UserService.java [hunk 1]
```diff
@@ -42,6 +43,8 @@
 @Path("/userServices/")
 public class UserService {
     private static final Log log = LogFactory.getLog(UserService.class);
+    int tenantId = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantId();
+    String str = String.valueOf(tenantId);
```
Comment (himasha) [inline components/bpmn/bpmn-rest/src/main/java/org/wso2/carbon/bpmn/rest/service/stats/UserService.java]: Add a meaningful variable name.

Comment (NatashaWso2) [inline components/bpmn/bpmn-rest/src/main/java/org/wso2/carbon/bpmn/rest/service/stats/UserService.java]: Fixed with commit ad06991fbc17e008e0864b1d55f7c6e58ee46200

File: components/bpmn/bpmn-rest/src/main/java/org/wso2/carbon/bpmn/rest/service/stats/UserService.java [hunk 2]
```diff
@@ -80,20 +83,20 @@ public ResponseHolder getNoOfTasksCompletedByUser() throws UserStoreException {
...
 
-            long count = BPMNOSGIService.getHistoryService()
-                    .createHistoricTaskInstanceQuery().taskAssignee(u).finished().count();
-            if (count == 0) {
-                userInfo.setTaskCount(0);
-            } else {
-                userInfo.setTaskCount(count);
+                long count = BPMNOSGIService.getHistoryService()
+                        .createHistoricTaskInstanceQuery().taskTenantId(str).taskAssignee(u).finished().count();
+                if (count == 0) {
```
Comment (vinok88) [inline components/bpmn/bpmn-rest/src/main/java/org/wso2/carbon/bpmn/rest/service/stats/UserService.java]: This if else is not required. you can directly use userInfo.setTaskCount(count);

File: components/bpmn/bpmn-rest/src/main/java/org/wso2/carbon/bpmn/rest/service/stats/UserService.java [hunk 3]
```diff
@@ -113,32 +116,32 @@ public ResponseHolder getAvgDurationForTasksCompletedByUser() throws UserStoreEx
...
+            String[] users = (String[]) getUserList().getData().toArray();
+            for (String u : users) {
+
+                UserTaskDuration userInfo = new UserTaskDuration();
+                userInfo.setUserName(u);
+
+                long count = BPMNOSGIService.getHistoryService()
+                        .createHistoricTaskInstanceQuery().taskTenantId(str).taskAssignee(u).finished().count();
+                if (count == 0) {
+                    double avgTime = 0;
```
Comment (vinok88) [inline components/bpmn/bpmn-rest/src/main/java/org/wso2/carbon/bpmn/rest/service/stats/UserService.java]: what is the point of using this avgTime variable? userInfo.setAvgTimeDuration(0); will do.

File: components/bpmn/bpmn-rest/src/main/java/org/wso2/carbon/bpmn/rest/service/stats/UserService.java (hunk 2 above)
Comment (NatashaWso2) [inline components/bpmn/bpmn-rest/src/main/java/org/wso2/carbon/bpmn/rest/service/stats/UserService.java]: Fixed with commit  a965c4a

File: components/bpmn/bpmn-rest/src/main/java/org/wso2/carbon/bpmn/rest/service/stats/UserService.java (hunk 3 above)
Comment (NatashaWso2) [inline components/bpmn/bpmn-rest/src/main/java/org/wso2/carbon/bpmn/rest/service/stats/UserService.java]: Fixed with commit a965c4a

**Pull request closed without merging**